homeassistant/components/intesishome/climate.py

```diff
@@ -0,0 +1,405 @@
+        """Copy values from controller dictionary to climate device."""
+        # Update values from controller's device dictionary
+        self._connected = self._controller.is_connected
+        self._current_temp = self._controller.get_temperature(self._device_id)
+        self._fan_speed = self._controller.get_fan_speed(self._device_id)
+        self._power = self._controller.is_on(self._device_id)
+        self._min_temp = self._controller.get_min_setpoint(self._device_id)
+        self._max_temp = self._controller.get_max_setpoint(self._device_id)
+        self._rssi = self._controller.get_rssi(self._device_id)
+        self._run_hours = self._controller.get_run_hours(self._device_id)
+        self._target_temp = self._controller.get_setpoint(self._device_id)
+        self._outdoor_temp = self._controller.get_outdoor_temperature(self._device_id)
+
+        # Operation mode
+        mode = self._controller.get_mode(self._device_id)
+        self._hvac_mode = MAP_IH_TO_HVAC_MODE.get(mode)
+
+        # Swing mode
+        # Climate module only supports one swing setting.
+        self._vvane = self._controller.get_vertical_swing(self._device_id)
+        self._hvane = self._controller.get_horizontal_swing(self._device_id)
+
+        if self._vvane == "swing" and self._hvane == "swing":
+            self._swing = SWING_BOTH
+        elif self._vvane == "swing":
+            self._swing = SWING_VERTICAL
+        elif self._hvane == "swing":
+            self._swing = SWING_HORIZONTAL
+        else:
+            self._swing = SWING_OFF
+
+    async def async_will_remove_from_hass(self):
+        """Shutdown the controller when the device is being removed."""
+        await self._controller.stop()
+
+    @property
+    def icon(self):
+        """Return the icon for the current state."""
+        icon = None
+        if self._power:
+            icon = MAP_STATE_ICONS.get(self._hvac_mode)
+        return icon
+
+    async def async_update_callback(self, device_id=None):
+        """Let HA know there has been an update from the controller."""
+        # Track changes in connection state
+        if not self._controller.is_connected and self._connected is True:
+            # Connection has dropped
+            self._connected = False
+            reconnect_minutes = 1 + randrange(10)
+            _LOGGER.error(
+                "Connection to %s API was lost. Reconnecting in %i minutes",
+                self._device_type,
+                reconnect_minutes,
+            )
+            # Schedule reconnection
+            async_call_later(
+                self.hass, reconnect_minutes * 60, self._controller.connect()
+            )
+
+        if self._controller.is_connected and self._connected is False:
+            # Connection has been restored
+            self._connected = True
+            _LOGGER.debug("Connection to %s API was restored.", self._device_type)
+
+        if self._device_id == str(device_id) or not device_id:
```

Comment:
Why do we need to copy to string? We don't seem to do that when setting the corresponding instance attribute.

Reply:
The library was sending an integer for device ID but platform uses string for device ID. I've updated the library to send a string here instead